pkg/service/association.go

```diff
@@ -58,22 +58,21 @@ func isDone(ctx context.Context) bool {
 }
 
 func ensureSetupPfcpAssociation(ctx context.Context, upf *smf_context.UPF, upfStr string) {
-	var alertTime time.Time
 	alertInterval := smf_context.SMF_Self().AssociationSetupFailedAlertInterval
 	for {
 		err := setupPfcpAssociation(upf, upfStr)
 		if err == nil {
 			return
 		}
-		now := time.Now()
-		if alertTime.IsZero() || now.After(alertTime.Add(alertInterval)) {
-			logger.AppLog.Errorf("Failed to setup an association with UPF%s, error:%+v", upfStr, err)
-			alertTime = now
-		}
-
-		if isDone(ctx) {
+		logger.AppLog.Errorf("Failed to setup an association with UPF%s, error:%+v", upfStr, err)
```

Comment:
hi @YoheiKonno , if we are here then it means that association failed for some reason and we wait interval of time as defined in the smf configuration. The wait is being used with 'select' in a similar manner as in `keepHeartbeatTo`. Once the association succeeds then it exist the loop. Is it ok ?

Reply:
hi @YoheiKonno , thanks for the clarification. I made the changes. Can you please review?
Thanks

Reply:
hello @aviweit,  
Thanks for the changes. Preventing busy loop part looks good to me, however, I have a request regarding logging.

I assume that Error level logs are noticed to operators.  So I would like Error level logs to be output at the first time or when the alert interval has passed, rather than every time when PFCP setup fails. And I would like the log level of other than that cases to be debug level or no logs, otherwise, when a UPF downs long time, huge retry logs would be output.

Thanks.

Reply:
Thanks, LGTM